pkg/reconciler/v1alpha1/taskrun/resources/cloudevent/cloud_event_controller.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2019 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloudevent
+
+import (
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	"go.uber.org/zap"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// InitializeCloudEvents initializes the CloudEvents part of the
+// TaskRunStatus from a slice of PipelineResources
+func InitializeCloudEvents(tr *v1alpha1.TaskRun, prs []*v1alpha1.PipelineResource) {
+	// FIXME(afrittoli) If there are no events this is run every time
+	if len(tr.Status.CloudEvents) == 0 {
+		var targets []string
+		for _, output := range prs {
+			if output.Spec.Type == v1alpha1.PipelineResourceTypeCloudEvent {
+				cer, _ := v1alpha1.NewCloudEventResource(output)
+				targets = append(targets, cer.TargetURI)
+			}
+		}
+		if len(targets) > 0 {
+			tr.Status.CloudEvents = cloudEventDeliveryFromTargets(targets)
+		}
+	}
+}
+
+func cloudEventDeliveryFromTargets(targets []string) []v1alpha1.CloudEventDelivery {
+	// len(nil slice) is 0
```

Comment:
hm im not sure this comment is needed?

Reply:
nice catch, it is definitely not, leftover...